### PR TITLE
Add cardano-crypto-class package with crypto type classes

### PR DESCRIPTION
**Pull request by ruhatch**
This is all copied from `ouroboros-consensus` and then adapted to use `cardano-binary` instead of `serialise`.

### CODE REVIEW

File: binary/src/Cardano/Binary/Serialize.hs [hunk 1]
```diff
@@ -70,6 +71,10 @@ serializeEncoding =
     -- sacred.
         strategy = Builder.safeStrategy 1024 4096
 
+serializeEncoding' :: Encoding -> ByteString
```
Comment (mgudemann) [inline binary/src/Cardano/Binary/Serialize.hs]: maybe add a comment / doc

File: binary/src/Cardano/Binary/ToCBOR.hs [hunk 2]
```diff
@@ -500,6 +504,50 @@ instance (ToCBOR a, ToCBOR b, ToCBOR c, ToCBOR d) => ToCBOR (a,b,c,d) where
...
+  toCBOR (a, b, c, d, e) =
+    E.encodeListLen 5
+      <> toCBOR a
+      <> toCBOR b
+      <> toCBOR c
+      <> toCBOR d
+      <> toCBOR e
+
+  encodedSizeExpr size _ =
+    1
```
Comment (mgudemann) [inline binary/src/Cardano/Binary/ToCBOR.hs]: is this for the `Word8` tag?

File: cardano-crypto-class/cardano-crypto-class.cabal [hunk 3]
```diff
@@ -0,0 +1,105 @@
+name:                cardano-crypto-class
+version:             2.0.0
+synopsis:            Type classes abstractign over cryptography primitives for Cardano
```
Comment (mgudemann) [inline cardano-crypto-class/cardano-crypto-class.cabal]:
```suggestion
synopsis:            Type classes abstracting over cryptography primitives for Cardano
```

File: cardano-crypto-class/LICENSE [hunk 4]
```diff
@@ -0,0 +1,20 @@
+Copyright (c) 2016 IOHK
```
Comment (mgudemann) [inline cardano-crypto-class/LICENSE]: different date from entry in `.cabal` file below

File: scripts/buildkite/rebuild.hs [hunk 5]
```diff
@@ -13,7 +13,7 @@ import Turtle
 -- | Run build and upload coverage information when successful
 main :: IO ()
 main = do
-  buildResult <- buildStep (Just ["--scenario=ContinuousIntegration"])
+  buildResult <- buildStep Nothing
```
Comment (mgudemann) [inline scripts/buildkite/rebuild.hs]: what is the result of this change ?

Comment (ruhatch) [reply scripts/buildkite/rebuild.hs]: Just removes a test flag that's not used in this package

File: binary/src/Cardano/Binary/ToCBOR.hs (hunk 2 above)
Comment (ruhatch) [reply binary/src/Cardano/Binary/ToCBOR.hs]: For the list len I believe

File: binary/src/Cardano/Binary/Serialize.hs (hunk 1 above)
Comment (ruhatch) [reply binary/src/Cardano/Binary/Serialize.hs]: Sure thing

**Pull request merged**